pos_lot_selection/static/src/js/EditListPopup.esm.js

```diff
@@ -1,19 +1,49 @@
 /** @odoo-module */
-
 /*
-    Copyright 2022 Camptocamp SA
+    Copyright 2023 Dixmit
+    Copyright 2022 Camptocamp
     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
 */
+
+import {onWillStart, useState} from "@odoo/owl";
+import {ConnectionLostError} from "@web/core/network/rpc_service";
+import {EditListInput} from "@point_of_sale/app/store/select_lot_popup/edit_list_input/edit_list_input";
 import {EditListPopup} from "@point_of_sale/app/store/select_lot_popup/select_lot_popup";
+
 import {_t} from "@web/core/l10n/translation";
 import {patch} from "@web/core/utils/patch";
-import {session} from "@web/session";
 
+patch(EditListInput.prototype, {
+    get_lot_name(lot) {
+        return lot.name;
+    },
+});
 patch(EditListPopup.prototype, {
     setup() {
-        super.setup(...arguments);
+        super.setup();
+
+        this.data = useState({lots: []});
+        onWillStart(this.onWillStart);
+    },
+    async onWillStart() {
         if (this.props.title === _t("Lot/Serial Number(s) Required")) {
-            this.props.lots = session.lots;
+            // We keep this in order to ensure that this call is only done
+            // when we add a serial
+            try {
+                this.data.lots = await this.env.services.orm.call(
```

Comment:
Hi @etobella, 
It always fetch the data from server. it does not work in offline mode.
![image](https://github.com/nguyenminhchien/pos/assets/8724793/f57c61a3-d4db-4489-8603-47afb69b29cd)

Reply:
obviously it is not getting the data, but no error is raised and it will work properly, actually, an error is raised with the problem of connection

Reply:
I'm ok to merge this commit, but its title "[REF] pos_lot_selection: Make it work with Offline mode" is misleading. so you have 2 options:
1. Make it work in offline mode properly, or.
2. Change your commit title to describe your work correctly.

I appreciate your contribution, thank you.

Reply:
Well, actually, it is making it work in offline mode. ON the previous version, it was not working on offline mode (an error was raised and it was failing) right now it works, but the functionality is limited on offline mode (as it should be....)

I will change the commit description to `[REF] pos_lot_selection: Make it not fail on Offline mode. Make it work with limited functionality`

Reply:
@etobella I also think it would be better to download the lots in the pos to allow selection in offline mode
Can you do that please ? @nguyenminhchien I'm sponsoring ;-)

Reply:
I made a change to make it work properly on Offline mode.

Can you review @nguyenminhchien ?

If you like it, I will squash all the commits on a single one

Reply:
yes please. i will keep your squashed commit in the PR to oca. thanks.

Reply:
Thanks @etobella